src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/ODataServiceDocumentResourceV2Provider.cs

```diff
@@ -119,7 +119,7 @@ public ODataServiceDocumentResourceV2Provider()
             }
 
             string serviceDocumentBaseAddress = null;
-            if (response.Status != HttpSourceResultStatus.NotFound)
```

Comment:
They should fail on a blank response when parsing it, the same as they did before.